src/scrcmd_move_tutor.c

```diff
@@ -0,0 +1,335 @@
+        if (sTutorMoves[i].move == move) {
+            *result = sTutorMoves[i].cost;
+            return FALSE;
+        }
+    }
+    GF_ASSERT(FALSE);
+    *result = 0;
+    return FALSE;
+}
+
+// check if mon at given slot can learn from headbutt move tutor
+BOOL ScrCmd_656(ScriptContext *ctx) {
+    u16 slot = ScriptGetVar(ctx);
+    u16 *result = ScriptGetVarPointer(ctx);
+    PARTY *party = SaveArray_PlayerParty_Get(ctx->fsys->savedata);
+    Pokemon *mon = GetPartyMonByIndex(party, slot);
+    u32 numLearnableMoves = GetLearnableTutorMoves(mon, MOVE_TUTOR_NPC_HEADBUTT, NULL);
+    *result = numLearnableMoves > 0 ? TRUE : FALSE;
+    return FALSE;
+}
+
+static u16 GetMoveTutorLearnsetIndex(u16 species, u8 form) {
+    u16 index;
+    // subtract 2 because SPECIES_EGG and SPECIES_BAD_EGG are missing
+    switch (species) {
+        case SPECIES_DEOXYS:
+            if (form == DEOXYS_ATTACK) {
+                index = SPECIES_DEOXYS_ATK - 2;
+            } else if (form == DEOXYS_DEFENSE) {
+                index = SPECIES_DEOXYS_DEF - 2;
+            } else if (form == DEOXYS_SPEED) {
+                index = SPECIES_DEOXYS_SPD - 2;
+            }
+            break;
+        case SPECIES_WORMADAM:
+            if (form == WORMADAM_SANDY) {
+                index = SPECIES_WORMADAM_SANDY - 2;
+            } else if (form == WORMADAM_TRASH) {
+                index = SPECIES_WORMADAM_TRASH - 2;
+            }
+            break;
+        case SPECIES_GIRATINA:
+            if (form == GIRATINA_ORIGIN) {
+                index = SPECIES_GIRATINA_ORIGIN - 2;
+            }
+            break;
+        case SPECIES_SHAYMIN:
+            if (form == SHAYMIN_SKY) {
+                index = SPECIES_SHAYMIN_SKY - 2;
+            }
+            break;
+        case SPECIES_ROTOM:
+            if (form == ROTOM_HEAT) {
+                index = SPECIES_ROTOM_HEAT - 2;
+            } else if (form == ROTOM_WASH) {
+                index = SPECIES_ROTOM_WASH - 2;
+            } else if (form == ROTOM_FROST) {
+                index = SPECIES_ROTOM_FROST - 2;
+            } else if (form == ROTOM_FAN) {
+                index = SPECIES_ROTOM_FAN - 2;
+            } else if (form == ROTOM_MOW) {
+                index = SPECIES_ROTOM_MOW - 2;
+            }
+            break;
+    }
+    return index - 1;
+}
+
+static u16 GetLearnableTutorMoves(Pokemon *mon, u32 moveTutorNpc, u8 dest[]) {
+    s32 i, j;
+    u16 currentMoves[MAX_MON_MOVES];
+    for (i = 0; i < MAX_MON_MOVES; i++) {
+        currentMoves[i] = GetMonData(mon, MON_DATA_MOVE1 + i, NULL);
+    }
+    u32 species = GetMonData(mon, MON_DATA_SPECIES, NULL);
+    u32 form = GetMonData(mon, MON_DATA_FORM, NULL);
+    MoveTutorLearnset *learnset = GetMoveTutorLearnset(HEAP_ID_FIELD, GetMoveTutorLearnsetIndex(species, form));
+    u16 numLearnableMoves = 0;
+    for (j = 0; j < NELEMS(sTutorMoves); j++) {
+        BOOL canLearnMove = (learnset->moves[j / 8] >> (j % 8)) & 1;
```

Comment:
this maths is a bit... arbritrary? might be an idea to explain it? (if you can figure it out yourself of course)